morango/sync/operations.py

```diff
@@ -580,39 +604,41 @@ def _queue_into_buffer_v2(transfersession):
                 )
                 + ")"
             )
-    partition_conditions = [_join_with_logical_operator(partition_conditions, "OR")]
 
-    # combine conditions and filter by profile
-    where_condition = _join_with_logical_operator(
-        profile_condition + partition_conditions, "AND"
-    )
+        partition_conditions = [_join_with_logical_operator(partition_conditions, "OR")]
 
-    # execute raw sql to take all records that match condition, to be put into buffer for transfer
-    select_buffer_query = (
-        """SELECT DISTINCT
-                id, serialized, deleted, last_saved_instance, last_saved_counter, hard_deleted, model_name, profile,
-                partition, source_id, conflicting_serialized_data,
-                CAST ('{transfer_session_id}' AS {transfer_session_id_type}), _self_ref_fk
-            FROM {store} WHERE {condition}
-        """.format(
-            transfer_session_id=transfersession.id,
-            transfer_session_id_type=TransferSession._meta.pk.rel_db_type(connection),
-            condition=where_condition,
-            store=Store._meta.db_table,
+        # combine conditions and filter by profile
+        where_condition = _join_with_logical_operator(
+            profile_condition + partition_conditions, "AND"
         )
-    )
+
+        # execute raw sql to take all records that match condition, to be put into buffer for transfer
+        select_buffers.append(
+            """SELECT DISTINCT
```

Comment:
The `DISTINCT` shouldn't be needed here, but it shouldn't actually do anything either.

Reply:
Good point. Duplication is more likely to occur _across_ SELECTs, not _within_ one of them, but even there we're probably safe now that we remove redundancy in the FSIC. I'll remove it.

Reply:
Removed -- let's see what the tests think of that